test/stateDirectivesSpec.js

```diff
@@ -291,6 +291,71 @@ describe('uiStateRef', function() {
     }));
   });
 
+  describe('links with dynamic state definitions', function () {
+    var template;
+
+    beforeEach(inject(function($rootScope, $compile, $state) {
+      el = angular.element('<a ui-state="state" ui-state-params="params">state</a>');
+      scope = $rootScope;
+      angular.extend(scope, { state: 'contacts', params: {} });
+      template = $compile(el)(scope);
+      scope.$digest();
+    }));
+
+    it('sets the correct initial href', function () {
+      expect(angular.element(template[0]).attr('href')).toBe('#/contacts');
+    });
+
+    it('updates to the new href', function () {
+      expect(angular.element(template[0]).attr('href')).toBe('#/contacts');
+
+      scope.state = 'contacts.item';
+      scope.params = { id: 5 };
```

Comment:
This looks way better to me than what I previously had; https://github.com/kasperlewau/ui-router/commit/110d793ee6591b139d390f49c642edc93326fe74#diff-e571eaac1af787a06dfe6dfe7210a4d9R312. 

Specifying the state ref *including* the params as a string felt sort of wonky when all things come around. Makes more sense to split it. :+1: